src/common/store-data-to-scan-node-result-converter.ts

```diff
@@ -95,10 +94,9 @@ export function convertAssessmentStoreDataToScanNodeResults(
     ruleInfoMap?: ScannerRuleInfoMap,
 ): ScanNodeResult[] | null {
     if (
-        isNullOrUndefined(assessmentStoreData) ||
-        isNullOrUndefined(assessmentStoreData.assessments) ||
-        isNullOrUndefined(selectedTest) ||
-        isNullOrUndefined(assessmentStoreData.assessments[selectedTest])
+        selectedTest == null ||
+        assessmentStoreData?.assessments == null ||
+        assessmentStoreData?.assessments[selectedTest] == null
```

Comment:
I tried `assessmentStoreData?.assessments?[selectedTest] = null`, but it produces the following message: "This condition will always return false since JavaScript compares objects to reference, not by value. ts(2839)"

Reply:
That error is happening because it's intepreting `assessmentStoreData?.assessments` and `[selectedTest] == null` with a `?` in between as the first two clauses of a ternary expression; what you'd instead want here is the optional chaining operator (`?.` with a period after) in between those clauses, like this:

`assessmentStoreData?.assessments?.[selectedTest] == null`

That said, I think the version as written here is at least as clear to read, I'm fine with leaving it as-is if you'd prefer.